app/controllers/admin/customers_controller.rb

```diff
@@ -53,6 +55,22 @@ def destroy
       end
     end
 
+    # GET /admin/customers/:id/addresses
+    # Used by standing orders form to load details for selected customer
+    def addresses
+      finder = OpenFoodNetwork::AddressFinder.new(@customer, @customer.email)
+      bill_address = Api::AddressSerializer.new(finder.bill_address).serializable_hash
+      ship_address = Api::AddressSerializer.new(finder.ship_address).serializable_hash
+      render json: { bill_address: bill_address, ship_address: ship_address }
+    end
+
+    # GET /admin/customers/:id/cards
+    # Used by standing orders form to load details for selected customer
+    def cards
+      cards = Spree::CreditCard.where(user_id: @customer.user_id)
+      render json: ActiveModel::ArraySerializer.new(cards, each_serializer: Api::CreditCardSerializer)
+    end
+
```

Comment:
have you considered moving these to a separate controller or simply having standing orders as a new REST resource? from my experience, when we diverge from REST in the controllers, things go wild pretty soon. Mainly because the class drops its cohesion.

Reply:
Happy to put these two actions in their own controllers. The reason they are here is that they don't really belong in the `AddressesController` or `CreditCardsController`, and I was trying to avoid creating a `CustomerAddressesController` and a `CustomerCreditCardsController`.

Reply:
Does make sense then, a `StandingOrdersController`? If so, I would just create an issue for it as it would lead to a quite a big refactor.

Reply:
My [StandingOrdersController](https://github.com/oeoeaio/openfoodnetwork/blob/standing-orders/app/controllers/admin/standing_orders_controller.rb) is already pretty full, and IMHO these actions belong there much less than they belong here in `CustomersController`.

Reply:
Ok, I thought it didn't exist yet...